### DISCUSSION TITLE
How to send and receive data from(to) Thread devices outside the home mesh network?

### DISCUSSION THREAD

**Discussion by ValNyrov** [openthread/openthread#8305] [Q&A] [answered]
Hello OpenThread Community!

I successfully completed the laboratory work - "Thread Border Router - Bidirectional IPv6 Connectivity and DNS-Based Service Discovery"  https://openthread.io/codelabs/openthread-border-router#0  (also successfully registered services). I can ping and send UDP packets to Thread devices inside my home network. I don't yet understand how to send and receive data from Thread devices outside my home network, and I need a hint-which technology to use for this purpose. Maybe it's CoAP, MQTT or there's something else? Please advise me what I need to study. Thank you.

Best Regards,
Valeriy

**Comment by jwhui** [comment, marked as answer]
Thread transports IPv6 datagrams and can support a number of IPv6-based application protocols, including Matter, CoAP, MQTT, and others. Each application layer has its benefits and which one to select largely depends on the use case. The ability to support a number of different application layer protocols is one of Thread's main benefits! Various vendors provide SDKs that offer different application samples running on top of Thread. I suggest taking a look at those examples.

**Reply by ValNyrov** [reply]
Thank you, can you give a couple of links to examples?

**Reply by jwhui** [reply]
Follow the links on our vendor support page: https://openthread.io/platforms
